### PR TITLE
fix: Set-ModuleFileAndFolderSetup pathway fix 

### LINKED ISSUE #2072: [AVM Question/Feedback]: Set-AVMModule.ps1 - bug
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Description

There is a bug in the PS script avm/utilities/tools/Set-AVMModule.ps1.
It internally invokes Set-ModuleFileAndFolderSetup.ps1 script.

The split operation in the file Set-ModuleFileAndFolderSetup.ps1 on line 52 does not work properly if the value in the $providerNamespace var appears earlier in the file path.

In my case
```
$FullModuleFolderPath = "C:\_DATA_L1\_Projects\_bicep-registry-modules\avm\ptn\data\private-analytical-workspace"
```
$providerNamespace is computed as "data"  but line 42 gets confused by the word “data” at the start of the file path ```"C:\_DATA_L1"```.

In other words, if the file path contains the value matching $providerNamespace value in multiple places, the code fails.

You can reproduce it:
```
$CurrentLevelFolderPath = ""
$FullModuleFolderPath = "C:\_DATA_L1\_Projects\_bicep-registry-modules\avm\ptn\data\private-analytical-workspace"


    if ([String]::IsNullOrEmpty($CurrentLevelFolderPath)) {
        # First invocation. Handling provider namespace
        $resourceTypeIdentifier = ($FullModuleFolderPath -split '[\/|\\]{1}avm[\/|\\]{1}(res|ptn)[\/|\\]{1}')[2] # avm/res/<provider>/<resourceType>
        $providerNamespace, $resourceType, $childResourceType = $resourceTypeIdentifier -split '[\/|\\]', 3
        
        $avmModuleRoot = ($FullModuleFolderPath -split $providerNamespace)[0]
        Write-Host "RESULT: $providerNamespace"
        Write-Host "RESULT: $avmModuleRoot"

        $currentLevelFolderPath = Join-Path $avmModuleRoot $providerNamespace $resourceType
    }
```

### PR DESCRIPTION
**Pull request by ChrisSidebotham**
## Description

Closes #2072 

Fixes bug in Set-ModuleFileAndFolderSetup which occurs when the pathway contains a duplicate folder name to the provider namespace

## Pipeline Reference

| Pipeline |
| -------- |
|      [avm.res.storage.storage-account](https://github.com/ChrisSidebotham/bicep-registry-modules/actions/workflows/avm.res.storage.storage-account.yml)    |

## Type of Change

- [x] Update to CI Environment or utlities (Non-module effecting changes)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/utilities/tools/helper/Set-ModuleFileAndFolderSetup.ps1
Comment (eriqua) [inline avm/utilities/tools/helper/Set-ModuleFileAndFolderSetup.ps1:39]: Is the tab size change intended? Asking because it makes the review slightly harder (from phone at least 😜)

Comment (ChrisSidebotham) [reply avm/utilities/tools/helper/Set-ModuleFileAndFolderSetup.ps1]: updated :)

**Comment by ChrisSidebotham** [comment]
@eriqua / @AlexanderSehr - Would you please be able to review this PR to unblock the linked issue

File: avm/utilities/tools/helper/Set-ModuleFileAndFolderSetup.ps1
Comment (eriqua) [reply avm/utilities/tools/helper/Set-ModuleFileAndFolderSetup.ps1]: Waiting for the environment to stabilize to avoid additional failed pipelines

File: avm/utilities/tools/helper/Set-ModuleFileAndFolderSetup.ps1
Comment (AlexanderSehr) [inline avm/utilities/tools/helper/Set-ModuleFileAndFolderSetup.ps1:50]: HEy @ChrisSidebotham, want to consider utls here too? Though to be fair I'd be curious to see if we'd have e.g. a `waf-aligned` test 😄

Comment (eriqua) [reply avm/utilities/tools/helper/Set-ModuleFileAndFolderSetup.ps1:50]: To be fair, we have many places where this is the case and utl are not introduced yet. I'd suggest to not mix that up and take it later for the whole repo as part of the PR adding their support

Comment (ChrisSidebotham) [reply avm/utilities/tools/helper/Set-ModuleFileAndFolderSetup.ps1:50]: @AlexanderSehr - Lets leave this to a wider set-avmmodule utl allignment PR?

Comment (AlexanderSehr) [reply avm/utilities/tools/helper/Set-ModuleFileAndFolderSetup.ps1:50]: I don't particularly care. Figured it would be good to immediately do it right here instead of knowingly adding outdated code, but not going to start a discussion over it if there are contrary opinons. Will resolve

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**
